### PR TITLE
EPMDJ-2546 Created env variable for agent ws timeout

### PR DESCRIPTION
**Pull request by ViktoriiaLebedeva**

### CODE REVIEW

File: core/src/main/kotlin/endpoints/agent/AgentWsExtension.kt [hunk 1]
```diff
@@ -51,15 +52,18 @@ class WsDeferred(asession: AgentWsSession, val clb: suspend () -> Unit, topicNam
...
 }
 
-
 open class AgentWsSession(val session: DefaultWebSocketServerSession) : DefaultWebSocketServerSession by session {
 
     val subscribers = ConcurrentHashMap<String, Signal>()
+    val timeout = application.drillConfig
+        .config("agents")
+        .config("socket")
+        .property("timeout").getString().toInt().seconds
```
Comment (LikhanovMaxim) [inline core/src/main/kotlin/endpoints/agent/AgentWsExtension.kt]: move the calculation into Application.kt

File: core/src/main/kotlin/endpoints/agent/AgentWsExtension.kt [hunk 2]
```diff
@@ -51,15 +52,15 @@ class WsDeferred(asession: AgentWsSession, val clb: suspend () -> Unit, topicNam
         return this
     }
 
-    suspend fun await() {
-        signal.await()
+    suspend fun await(timeout: Duration = asession.timeout) {
+        signal.await(timeout = timeout)
```
Comment (LikhanovMaxim) [inline core/src/main/kotlin/endpoints/agent/AgentWsExtension.kt]:
```suggestion
    suspend fun await() {
        signal.await(timeout = asession.timeout)
```

**Review by LikhanovMaxim — APPROVED** [review]

File: test-framework/src/main/kotlin/com/epam/drill/e2e/AppConfig.kt [hunk 3]
```diff
@@ -30,6 +30,7 @@ class AppConfig(var projectDir: File) {
             put("ktor.deployment.sslPort", sslPort)
             put("drill.devMode", "true")
             put("drill.plugins.remote.enabled", "false")
+            put("drill.agents.socket.timeout","90")
```
Comment (viktorbll) [inline test-framework/src/main/kotlin/com/epam/drill/e2e/AppConfig.kt]:
```suggestion
            put("drill.agents.socket.timeout", "90")
```

File: core/src/main/kotlin/config/Application.kt [hunk 4]
```diff
@@ -2,10 +2,16 @@ package com.epam.drill.admin.config
...
+import kotlin.time.*
 
 val Application.drillConfig: ApplicationConfig
     get() = environment.config.config("drill")
 
 
 val Application.isDevMode: Boolean
     get() = drillConfig.property("devMode").getString().toBoolean()
+
+val Application.agentsSocketTimeout: Duration
```
Comment (viktorbll) [inline core/src/main/kotlin/config/Application.kt]:
```suggestion
val Application.agentSocketTimeout: Duration
```

**Review by viktorbll — APPROVED** [review]

**Pull request merged**